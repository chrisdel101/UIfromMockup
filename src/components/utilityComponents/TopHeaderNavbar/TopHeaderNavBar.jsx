import React from 'react'
import Icon from '../Icon/Icon'
import { Nav, Navbar, NavDropdown } from 'react-bootstrap'
import styles from './index.css'

function renderLinkType(prop, i) {
  if (prop.type === 'link') {
    return (
      <Nav.Link key={i} href={`/${prop.text}`}>
        {prop.text}
      </Nav.Link>
    )
  } else if (prop.type === 'icon') {
    return (
      <Nav.Link key={i} href={`/${prop.text}`}>
        <Icon type={prop.iconName} />
      </Nav.Link>
    )
  }
}
function renderNavMarkup(props) {
  console.log(props)
  return (
    <React.Fragment>
      <Navbar
        className="top-navbar"
        collapseOnSelect
        expand="md"
        bg="dark"
        variant="dark"
      >
        <Navbar.Toggle aria-controls="responsive-navbar-nav" />
        <Navbar.Collapse id="responsive-navbar-nav">
          <Nav className="mr-auto">
            {props.links.map((link, i) => {
              return renderLinkType(link, i)
            })}
          </Nav>
          <Nav />
        </Navbar.Collapse>
      </Navbar>
    </React.Fragment>
  )
}
function TopNavBar(props) {
  return <React.Fragment>{renderNavMarkup(props)}</React.Fragment>
}

export default TopNavBar
