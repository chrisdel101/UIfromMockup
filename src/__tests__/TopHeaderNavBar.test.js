import React from 'react'
import TopHeaderNavBar from '../components/utilityComponents/TopHeaderNavBar/TopHeaderNavBar.jsx'
import renderer from 'react-test-renderer'

test('create and match snapshot', () => {
  const links = ['Intro', 'Work', 'About', 'Contacts']
  const component = renderer.create(<TopHeaderNavBar links={links} />)

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
