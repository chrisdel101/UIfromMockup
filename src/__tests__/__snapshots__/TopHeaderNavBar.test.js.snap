// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create and match snapshot 1`] = `
<nav
  className="top-navbar navbar navbar-expand-md navbar-dark"
>
  <button
    aria-controls="responsive-navbar-nav"
    aria-label="Toggle navigation"
    className="navbar-toggler collapsed"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    aria-expanded={null}
    className="navbar-collapse collapse show"
    id="responsive-navbar-nav"
  >
    <div
      className="mr-auto  navbar-nav"
      onKeyDown={[Function]}
    />
    <div
      className="navbar-nav"
      onKeyDown={[Function]}
    />
  </div>
</nav>
`;
